test/functional/Java11andUp/src/org/openj9/test/varhandle/TestVarHandleInfo.java

```diff
@@ -0,0 +1,115 @@
+//package java.lang.invoke;
+package org.openj9.test.varhandle;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandleInfo;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.reflect.Modifier;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.VarHandle;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+
+@Test(groups = { "level.extended" })
+public class TestVarHandleInfo {
+
+	protected static Logger logger = Logger.getLogger(TestVarHandleInfo.class);
+	private static final String TEST_VAR_HANDLE_INFO = "Lorg/openj9/test/varhandle/TestVarHandleInfo;"; //$NON-NLS-1$
+	private static final String INTEGER = "Ljava/lang/Integer;"; //$NON-NLS-1$
+	private static final int PUBLIC_FINAL_NATIVE = Modifier.PUBLIC + Modifier.FINAL + Modifier.NATIVE;
+	private static final int expectedModifiers = PUBLIC_FINAL_NATIVE;
+	private static final String EXPECTED_MODIFIERS_STRING = Modifier.toString(expectedModifiers);
+	private static final int expectedReferenceKind = MethodHandleInfo.REF_invokeVirtual;
+	
+	Lookup mylookup;
+	Integer instanceField; /* the target of the varhandle */
+	List<String> accessModes;
+	HashMap<String, MethodType> infos;
+
+	@Test
+	public void testAccessModes() throws NoSuchFieldException, IllegalAccessException {
+		VarHandle vh = mylookup.findVarHandle(TestVarHandleInfo.class, "instanceField", Integer.class); //$NON-NLS-1$
+		for (String methName: accessModes) {
+			logger.debug(methName);
+			VarHandle.AccessMode am = VarHandle.AccessMode.valueFromMethodName(methName);
+			MethodHandle mh = MethodHandles.varHandleExactInvoker(am,vh.accessModeType(am));
+			checkMethodHandleInfo(mylookup.revealDirect(mh), infos.get(methName), methName);
+
+			mh = MethodHandles.varHandleInvoker(am,vh.accessModeType(am));
+			checkMethodHandleInfo(mylookup.revealDirect(mh), infos.get(methName), methName);
+		}
+	}
+
+	private static void checkMethodHandleInfo(MethodHandleInfo info, MethodType expectedMethodType, String expectedMethodName) {	
+		assertEquals("Incorrect declaring class", VarHandle.class, info.getDeclaringClass()); //$NON-NLS-1$	
+
+		assertEquals("Incorrect methodType", expectedMethodType, info.getMethodType()); //$NON-NLS-1$
+		
+		int actualModifiers = info.getModifiers();
+		assertTrue("Missing modifiers: expected \""+EXPECTED_MODIFIERS_STRING  //$NON-NLS-1$
+				+ "\" (" + Integer.toHexString(expectedModifiers)  //$NON-NLS-1$
+		+ "), actual \"" + Modifier.toString(actualModifiers)  //$NON-NLS-1$
+		+ "\" ("+Integer.toHexString(actualModifiers)+")",  //$NON-NLS-1$ //$NON-NLS-2$
+		(expectedModifiers & actualModifiers) == expectedModifiers);
```

Comment:
Should this be an `assertEquals` between the expectedModifiers and the actualModifiers?  Why the `&`?

Reply:
> Should this be an `assertEquals` between the expectedModifiers and the actualModifiers?

Yes.  Will fix.

> Why the `&`?

The `actualModifiers` can have some extra (undocumented) bits set.